Added how to use Caddy Certificate for Postal

FlorinBuffet opened the pull request:
I had to find all of this information on different discussion threads until and saw that others also didn't find it. Therefor i added it to the official documentation.

sources:
https://github.com/orgs/postalserver/discussions/1572
https://github.com/orgs/postalserver/discussions/2673

FlorinBuffet committed:
Added how to use Caddy Certificate for Postal

sources:
https://github.com/orgs/postalserver/discussions/1572
https://github.com/orgs/postalserver/discussions/2673

willpower232 commented on content/3.features/smtp-tls.md:
````diff
@@ -21,6 +21,104 @@ You can use the command below to generate a self-signed certificate.
 
+### Using Caddy certificate for TLS
+
+#### Setting up Caddy to issue RSA certificate
+
+Caddy doesn't issue RSA certificates by default, so first we have to setup caddy to issue RSA certificate. This is done by adding following lines to the start of `/opt/postal/config/Caddyfile`:
+```yaml
+# Force Caddy to generate RSA certificates
+# https://github.com/postalserver/postal/discussions/1572#discussioncomment-1410343
+{
+  key_type rsa4096
+}
+```
````
Postal has supported non-RSA certificates for a couple of years now so this should be unnecessary https://github.com/postalserver/postal/pull/2126

FlorinBuffet replied:
With the default Caddy certificate it did not work unfortunately, but if you say this should be fixed i'll try again with a default certificate.

FlorinBuffet replied:
It works now, probably it was another config issue earlier.
Therefor i removed this part from the documentation.

willpower232 commented on content/3.features/smtp-tls.md:
```diff
@@ -21,6 +21,104 @@ You can use the command below to generate a self-signed certificate.
+    chmod o+r /opt/postal/config/smtp.*
+
+    # Restart Postal to use the new certificates
+    postal stop && sleep 15 && postal start
```
presumably `postal restart` is more efficient

FlorinBuffet replied:
fixed

willpower232 commented on content/3.features/smtp-tls.md:
````diff
@@ -21,6 +21,104 @@ You can use the command below to generate a self-signed certificate.
+##### Initial Manual Certificate Copy
+Before the monitoring script takes over, you should manually copy the certificates for the first time:
+```bash
+cp /opt/postal/caddy-data/caddy/certificates/acme.zerossl.com-v2-dv90/YOURDOMAIN/YOURDOMAIN.crt /opt/postal/config/smtp.cert
+cp /opt/postal/caddy-data/caddy/certificates/acme.zerossl.com-v2-dv90/YOURDOMAIN/YOURDOMAIN.key /opt/postal/config/smtp.key
````
these paths do not match the ones above

FlorinBuffet replied:
fixed, i didn't think of updating the path down here

FlorinBuffet committed:
Optimized Scripts

use postal restart instead of more complex postal stop && postal start
fixed directory in manual certificate copy

FlorinBuffet committed:
Removed unneccessary RSA4096 for Caddy

willpower232 commented on content/3.features/smtp-tls.md:
```diff
@@ -21,6 +21,93 @@ You can use the command below to generate a self-signed certificate.
+
+#### Setup automatic copying from Caddy to Postal
+
+To remove the need of the manual mainenance task to copy the certificate from Caddy to Postal we can automate this. The original discussion and author can be found [here](https://github.com/orgs/postalserver/discussions/2673).
```
```suggestion
To remove the need of the manual maintenance task to copy the certificate from Caddy to Postal, we can automate this. The original discussion and author can be found [here](https://github.com/orgs/postalserver/discussions/2673).
```

willpower232 commented on content/3.features/smtp-tls.md:
````diff
@@ -21,6 +21,93 @@ You can use the command below to generate a self-signed certificate.
+sudo systemctl enable monitor_certs.service
+sudo systemctl start monitor_certs.service
+```
+##### Initial Manual Certificate Copy
+Before the monitoring script takes over, you should manually copy the certificates for the first time:
````
```suggestion

##### Initial Manual Certificate Copy

Before the monitoring script takes over, you should manually copy the certificates for the first time:

```

willpower232 commented on content/3.features/smtp-tls.md:
````diff
@@ -21,6 +21,93 @@ You can use the command below to generate a self-signed certificate.
+```bash
+sudo systemctl daemon-reload
+```
+Enable and start the service:
````
```suggestion

Enable and start the service:

```

willpower232 commented on content/3.features/smtp-tls.md:
````diff
@@ -21,6 +21,93 @@ You can use the command below to generate a self-signed certificate.
+WantedBy=multi-user.target
+```
+
+##### Activate the Service
+Reload the systemd daemons:
````
```suggestion
##### Activate the Service

Reload the systemd daemons:

```

willpower232 commented on content/3.features/smtp-tls.md:
````diff
@@ -21,6 +21,93 @@ You can use the command below to generate a self-signed certificate.
+```bash
+sudo nano /etc/systemd/system/monitor_certs.service
+```
+Insert the following content:
````
```suggestion

Insert the following content:

```

willpower232 commented on content/3.features/smtp-tls.md:
````diff
@@ -21,6 +21,93 @@ You can use the command below to generate a self-signed certificate.
+```bash
+chmod +x /opt/postal/monitor_certs.sh
+```
+##### Create a systemd Service
+
+Make a systemd service file:
````
```suggestion

##### Create a systemd Service

Make a systemd service file:

```

willpower232 commented on content/3.features/smtp-tls.md:
````diff
@@ -21,6 +21,93 @@ You can use the command below to generate a self-signed certificate.
+```bash
+nano /opt/postal/monitor_certs.sh
+```
+Add following code to the script file:
````
```suggestion

Add following code to the script file:

```

willpower232 commented on content/3.features/smtp-tls.md:
```diff
@@ -21,6 +21,93 @@ You can use the command below to generate a self-signed certificate.
+
+##### Create Monitoring Script
+
+Create a script named `monitor_certs.sh`:
```
```suggestion
Create a script named `monitor_certs.sh`:

```

willpower232 committed:
Apply suggestions from code review

willpower232 commented on content/3.features/smtp-tls.md:
````diff
@@ -21,6 +21,107 @@ You can use the command below to generate a self-signed certificate.
+```bash
+#!/bin/bash
+
+CERT_DIR="/opt/postal/caddy-data/caddy/certificates/acme-v02.api.letsencrypt.org-directory/YOURDOMAIN/"
````
does this only work if you have caddy installed directly on the server or are you mapping a docker volume to get this out of the container?

FlorinBuffet replied:
yes this is for a direct install on the server, if it is in a docker volume, you'd probably have to adjust the paths

willpower232 replied:
having just re-familiarised myself with the docs https://docs.postalserver.io/getting-started/installation#caddy

looks like you have used the docker paths right?

FlorinBuffet replied:
As yes, of course, wherever possible i used the docker installation path and information
